hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/SigTransactionMetadataBuilder.java

```diff
@@ -13,77 +13,106 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.hedera.node.app.service.mono;
+package com.hedera.node.app.spi.meta;
 
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_PAYER_ACCOUNT_ID;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
 
-import com.hedera.node.app.service.mono.token.impl.AccountStore;
-import com.hedera.node.app.service.mono.token.impl.KeyOrLookupFailureReason;
+import com.hedera.node.app.spi.AccountKeyLookup;
+import com.hedera.node.app.spi.KeyOrLookupFailureReason;
 import com.hedera.node.app.spi.key.HederaKey;
-import com.hedera.node.app.spi.meta.TransactionMetadata;
 import com.hederahashgraph.api.proto.java.AccountID;
 import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
 import com.hederahashgraph.api.proto.java.TransactionBody;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import edu.umd.cs.findbugs.annotations.Nullable;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
 
 /**
- * Metadata collected when transactions are handled as part of "pre-handle" needed for signature
- * verification. This class may have subclasses in the future.
- *
- * <p>NOTE: This class shouldn't exist here, and is something of a puzzle. We cannot add it to SPI,
- * because it includes a dependency on AccountStore. But we also cannot put it in the app module,
- * because doing so would cause service modules to have a circular dependency on the app module.
- * Maybe we need some kind of base module from which services can extend and put it there?
+ * Builds {@link SigTransactionMetadata} by collecting needed information collected when
+ * transactions are handled as part of "pre-handle" needed for signature verification. This class
+ * will be subclassed For e.g., we need a {@link TransactionMetadata} with an inner {@link
+ * TransactionMetadata} for schedule transactions.
  */
-public class SigTransactionMetadata implements TransactionMetadata {
-    protected List<HederaKey> requiredKeys = new ArrayList<>();
+public class SigTransactionMetadataBuilder {
+    protected final List<HederaKey> requiredKeys = new ArrayList<>();
+    protected ResponseCodeEnum status = OK;
+
     protected TransactionBody txn;
-    protected AccountStore store;
+    protected final AccountKeyLookup keyLookup;
     protected AccountID payer;
 
-    protected ResponseCodeEnum status = OK;
+    public SigTransactionMetadataBuilder(final AccountKeyLookup keyLookup) {
```

Comment:
```suggestion
    public SigTransactionMetadataBuilder(@NonNull final AccountKeyLookup keyLookup) {
```

Reply:
Added annotations for other methods as well. Thanks